frontend/src/pages/modelServing/screens/projects/InferenceServiceModal/DataConnectionExistingField.tsx

```diff
@@ -0,0 +1,76 @@
+import * as React from 'react';
+import {
+  FormGroup,
+  Select,
+  SelectOption,
+  Stack,
+  StackItem,
+  TextInput,
+} from '@patternfly/react-core';
+import { DataConnection, UpdateObjectAtPropAndValue } from 'pages/projects/types';
+import { CreatingInferenceServiceObject } from '../../types';
+import { getDataConnectionDisplayName } from 'pages/projects/screens/detail/data-connections/utils';
+import '../../../../projects/screens/detail/storage/ManageStorageModal.scss';
+
+type DataConnectionExistingFieldType = {
+  data: CreatingInferenceServiceObject;
+  setData: UpdateObjectAtPropAndValue<CreatingInferenceServiceObject>;
+  dataConnections: DataConnection[];
+};
+
+const DataConnectionExistingField: React.FC<DataConnectionExistingFieldType> = ({
+  data,
+  setData,
+  dataConnections,
+}) => {
+  const [isOpen, setOpen] = React.useState<boolean>(false);
```

Comment:
FWIW, you never need to type of a `useState` when it's a primitive (0, false, '', etc...). When it's null/undefined or `{}` / `[]` you need to type it. Especially if you do not set a default.

No harm here, just FYI.

Reply:
Ok, yes, you're completely right, type inference with primitives is granted so typing is not mandatory in these scenarios. I just did it since we have several appearance with usestate and the type with primitives. But it's as weird as having something as `let stringVar: string = "hello world"`